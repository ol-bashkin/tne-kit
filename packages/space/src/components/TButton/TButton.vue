<template>
  <button :class="[classes.button]" @click="$emit('click', $event)" @mouseover="hover">
    <span :class="[classes.text]">
      <slot>@tne-kit</slot>
    </span>
  </button>
</template>

<script lang="ts" setup>
import { withDefaults, computed } from "vue";

interface Props {
  theme?: string;
  color?: string;
  fill?: string;
  size?: string;
}

const props = withDefaults(defineProps<Props>(), {
  theme: "light",
  color: "primary--base",
  fill: "secondary--base",
  size: "lg",
});

const emit = defineEmits(["click"]);

const classes = computed(() => {
  return {
    button: `t-button t-button--${props.size} ${props.fill}`,
    text: `t-text font-secondary text-${props.color}`,
  };
});

const hover = (event: any) => {
  console.log("hover");
};
</script>

<style lang="scss">
/* === Button Sizes === */
$sizes: (
  xl,
  lg,
  md,
  sm,
  xs) !default;

.t-button {
  border: none;

  @each $size in $sizes {
    &--#{$size} {
      height: var(--height-#{$size}
    );
    min-height: var(--height-#{$size});
    padding-left: var(--space-#{$size});
    padding-right: var(--space-#{$size});
    border-radius: var(--border-radius-#{$size});
    min-width: var(--width-#{$size});

    .t-text {
      font-size: var(--font-size-#{$size});
    }
  }
}

&:hover {
  cursor: pointer;
}
}
</style>
