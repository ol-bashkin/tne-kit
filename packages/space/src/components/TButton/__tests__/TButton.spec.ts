import { mount } from "@vue/test-utils";
import TButton from "../TButton.vue";

describe("WButton.vue", () => {
  test("should have an t-button class", () => {
    const wrapper = mount(TButton, {
      propsData: {
        theme: "light",
      },
    });
    expect(wrapper.classes()).toContain("t-button");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
