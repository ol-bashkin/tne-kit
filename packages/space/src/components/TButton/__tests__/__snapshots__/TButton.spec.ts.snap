// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WButton.vue should have an t-button class 1`] = `
<button class="t-button t-button--lg secondary--base">
  <span class="t-text font-secondary text-primary--base">@tne-kit</span>
</button>
`;
